src/tests/Microsoft.PowerFx.Core.Tests/UserDefinedFunctionTests.cs

```diff
@@ -153,5 +153,20 @@ public void TestCoercionWithUDFBody(string udfScript, string expectedIR)
 
             Assert.Equal(expectedIR, actualIR);
         }
+
+        [Theory]
+        [InlineData("/* Comment1 */ Foo(x: Number): Number = /* Comment2 */ Abs(x) /* Comment3 */;/* Comment4 */", 4)]
+        [InlineData("Foo(x: Number): Number /* Comment1 */ =  Abs(x);// Comment2", 2)]
+        public void TestCommentsFromUserDefinitionsScript(string script, int udfCount)
+        {
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false
+            };
+
+            var parseResult = UserDefinitions.Parse(script, parserOptions);
+
+            Assert.Equal(udfCount, parseResult.Comments.Count());
```

Comment:
We should have at least one test where we validate the span of the comments to make sure that they are correct.